apps/src/code-studio/components/libraries/LibraryCreationDialog.jsx

```diff
@@ -125,66 +146,94 @@ class LibraryCreationDialog extends React.Component {
   };
 
   displayFunctions = () => {
-    if (!this.state.loadingFinished) {
+    if (this.state.loadingState === LoadingState.LOADING) {
       return (
         <div style={styles.centerSpinner}>
           <Spinner />
         </div>
       );
+    } else if (this.state.loadingState === LoadingState.DONE_LOADING) {
```

Comment:
Given how spread out these if/elseif/else statements are, would it make sense to move the contents into their own functions?